### PR TITLE
BUG: fix reducing numpy ufuncs for Series/Index

### LINKED ISSUE #23312: BUG: reducing ufuncs don't return a scalar
But are wrapped again in the Series or Index:

```
In [1]: s = pd.Series([1, 2, 3])

In [3]: np.add.reduce(s)
Out[3]: 
0    6
1    6
2    6
dtype: int64

In [4]: np.add.reduce(s.values)
Out[4]: 6
```

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
closes #23312

### CODE REVIEW

File: pandas/tests/arithmetic/test_numeric.py [hunk 1]
```diff
@@ -833,6 +833,15 @@ def test_ufunc_coercions(self, holder):
         exp = tm.box_expected(exp, box)
         tm.assert_equal(result, exp)
 
+    @pytest.mark.parametrize('holder', [pd.Int64Index, pd.UInt64Index,
```
Comment (jreback) [inline pandas/tests/arithmetic/test_numeric.py]: holder -> box

File: pandas/tests/arithmetic/test_numeric.py [hunk 2]
```diff
@@ -833,6 +833,15 @@ def test_ufunc_coercions(self, holder):
...
+                                        pd.Float64Index, pd.Series])
+    def test_ufunc_reduce(self, holder):
+        idx = holder([1, 2, 3, 4, 5], name='x')
+
+        result = np.add.reduce(idx)
```
Comment (jreback) [inline pandas/tests/arithmetic/test_numeric.py]: use paramterize over some more ops here

**Review by jreback — CHANGES_REQUESTED** [review]
needs a whatsnew note

File: pandas/tests/arithmetic/test_numeric.py (hunk 2 above)
Comment (jorisvandenbossche) [reply pandas/tests/arithmetic/test_numeric.py]: We would only be testing that those work on the underlying numpy array (which they do :)), so I don't think it is that important

File: pandas/tests/arithmetic/test_numeric.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/tests/arithmetic/test_numeric.py]: The tests above also use holder, so I wanted to be consistent, but can change them all

Comment (jreback) [reply pandas/tests/arithmetic/test_numeric.py]: yes, we have settled on box a while back

File: pandas/tests/arithmetic/test_numeric.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/arithmetic/test_numeric.py]: yes exactly

**Comment by jreback** [comment]
needs a rebase

**Comment by jreback** [comment]
@jorisvandenbossche can you merge master and update

**Comment by jreback** [comment]
can you merge master (and move notes to 0.25) / or close if going to handle in ``__array_ufunc__``

**Pull request closed without merging**